DataFormats/TPC/include/DataFormatsTPC/ClusterHardware.h

```diff
@@ -0,0 +1,46 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file ClusterHardware.h
+/// \brief Class of a TPC cluster as produced by the hardware cluster finder (needs a postprocessing step to convert into ClusterNative
+/// \author David Rohr
+#ifndef ALICEO2_DATAFORMATSTPC_CLUSTERHARDWARE_H
+#define ALICEO2_DATAFORMATSTPC_CLUSTERHARDWARE_H
+
+namespace o2 { namespace DataFormat { namespace TPC{
+
+struct ClusterHardware { //Temporary draft of hardware clusters. The ...Pre members are yet to be defined, and will most likely either be floats or fixed point integers.
+  float mPadPre;                //< Quantity needed to compute the pad
+  float mTimePre;               //< Quantity needed to compute the time
+  float mSigmaPad2Pre;          //< Quantity needed to compute the sigma^2 of the pad
+  float mSigmaTime2Pre;         //< Quantity needed to compute the sigma^2 of the time
+  unsigned short mQMax;         //< QMax of the cluster
+  unsigned short mQTot;         //< Total charge of the cluster
+  unsigned char mRow;           //< Row of the cluster (local, needs to add PadRegionInfo::getGlobalRowOffset
+  unsigned char mFlags;         //< Flags of the cluster
+  
+  float getPad() const {return mPadPre / mQTot;}
+  float getTimeLocal() const {return mTimePre / mQTot;} //Returns the local time, not taking into accound the time bin offset of the container
+  float getSigmaPad2() const {return (mSigmaPad2Pre - mPadPre * mPadPre) / (mQTot * mQTot);}
+  float getSigmaTime2() const {return (mSigmaTime2Pre - mTimePre * mTimePre) / (mQTot * mQTot);}
+};
+
+struct ClusterHardwareContainer { //Temporary struct to hold a set of hardware clusters, prepended by an RDH, and a short header with metadata, to be replace
+                                  //The total structure is supposed to use up to 8 kb (like a readout block, thus it can hold up to 339 clusters ((8192 - 40) / 24)
+    unsigned long long int mRDH[8];   //< 8 * 64 bit RDH (raw data header)
```

Comment:
Hi, maybe it would be better to use (u-)int64_t when mapping to binary buffers with a fixed layout - long long int has *at least* 64 bits, int64_t has *exactly* 64bits. Similarly i'd go for int16_t instead of short etc.

Reply:
done, although I doubt that this will change in the future